src/server/svr_jobfunc.c

```diff
@@ -5290,6 +5295,60 @@ svr_clean_job_history(struct work_task *pwt)
 		}
 		/* restore the saved next in pjob */
 		pjob = nxpjob;
+
+		/* check if we spent too long hogging the pbs_server process here */
+		end_time = time(NULL);
+		if ((end_time - begin_time) > SVR_CLEAN_JOBHIST_SECS) {
+			/* Apparently the interval between history purges is too long */
+			/* reduce it using factor 0.7                                 */
+			time_between_tasks = (floor((double)time_between_tasks * 0.7));
+
+			/* no use reducing to less than 4 * SVR_CLEAN_JOBHIST_SECS */
+			/* since we'll already schedule a continuation task here */
```

Comment:
I don't understand this.  Where does the 4 * come into play?  I don't see 4 * anywhere else in the code change.

Reply:
4 is an arbitrary choice here based on the value of SVR_CLEAN_JOBHIST_SECS (which is 5 seconds). So we need not reduce the time between two tasks to less than 20 seconds.